integration-tests/it-consumer/src/main/java/org/apache/servicecomb/it/testcase/thirdparty/Test3rdPartyInvocation.java

```diff
@@ -0,0 +1,244 @@
+    dataTypeJaxrsSchemaAsync.add(5, 6).whenComplete((result, t) -> {
+      Assert.assertEquals(11, result.intValue());
+      Assert.assertNull(t);
+      addChecked.value = true;
+    }).get();
+    Assert.assertTrue(addChecked.value);
+
+    Holder<Boolean> postStringChecked = new Holder<>(false);
+    dataTypeJaxrsSchemaAsync.stringBody("abc").whenComplete((result, t) -> {
+      Assert.assertEquals("abc", result);
+      Assert.assertNull(t);
+      postStringChecked.value = true;
+    }).get();
+    Assert.assertTrue(postStringChecked.value);
+
+    Holder<Boolean> concatChecked = new Holder<>(false);
+    dataTypeJaxrsSchemaAsync.stringConcat("uvw", "xyz").whenComplete((result, t) -> {
+      Assert.assertEquals("uvwxyz", result);
+      Assert.assertNull(t);
+      concatChecked.value = true;
+    }).get();
+    Assert.assertTrue(concatChecked.value);
+  }
+
+  @Test
+  public void testSyncInvoke_RestTemplate() {
+    RestTemplate restTemplate = RestTemplateBuilder.create();
+    ResponseEntity<Integer> responseEntity = restTemplate
+        .getForEntity(
+            "cse://" + THIRD_PARTY_MICROSERVICE_NAME + "/rest/it-producer/v1/dataTypeJaxrs/add?num1=11&num2=22",
+            int.class);
+    Assert.assertEquals(200, responseEntity.getStatusCodeValue());
+    Assert.assertEquals(33, responseEntity.getBody().intValue());
+
+    ResponseEntity<String> stringBodyResponse = restTemplate
+        .exchange("cse://" + THIRD_PARTY_MICROSERVICE_NAME + "/rest/it-producer/v1/dataTypeJaxrs/stringBody",
+            HttpMethod.POST,
+            new HttpEntity<>("abc"), String.class);
+    Assert.assertEquals(200, stringBodyResponse.getStatusCodeValue());
+    Assert.assertEquals("abc", stringBodyResponse.getBody());
+  }
+
+  @Test
+  public void testAsyncInvoke_RestTemplate() throws ExecutionException, InterruptedException {
+    CseAsyncRestTemplate cseAsyncRestTemplate = new CseAsyncRestTemplate();
+    ListenableFuture<ResponseEntity<Integer>> responseFuture = cseAsyncRestTemplate
+        .getForEntity(
+            "cse://" + ASYNC_THIRD_PARTY_MICROSERVICE_NAME + "/rest/it-producer/v1/dataTypeJaxrs/add?num1=11&num2=22",
+            Integer.class);
+    ResponseEntity<Integer> responseEntity = responseFuture.get();
+    Assert.assertEquals(200, responseEntity.getStatusCodeValue());
+    Assert.assertEquals(33, responseEntity.getBody().intValue());
+
+    ListenableFuture<ResponseEntity<String>> stringBodyFuture = cseAsyncRestTemplate
+        .exchange("cse://" + ASYNC_THIRD_PARTY_MICROSERVICE_NAME + "/rest/it-producer/v1/dataTypeJaxrs/stringBody",
+            HttpMethod.POST,
+            new HttpEntity<>("abc"), String.class);
+    ResponseEntity<String> stringBodyResponse = stringBodyFuture.get();
+    Assert.assertEquals(200, stringBodyResponse.getStatusCodeValue());
+    Assert.assertEquals("abc", stringBodyResponse.getBody());
+  }
+
+  @Path("/rest/it-producer/v1/dataTypeJaxrs")
```

Comment:
for different deploy mode, target microservice name is different.
so "it-producer" can not hard code
based on zhengyangyong's PR, if that one merged first, then this one will failed.

Reply:
I've checked zhengyangyong's commits and test it by rebasing those commits into my branch. There is no problem with the producer's service name because other deploy mode will not trigger this test.
The only problem is the initialization method of `GateRestTemplate` is changed. And I've changed to use new initialization method in new commit.